src/components/Email/EmailVerificationStatus.tsx

```diff
@@ -0,0 +1,149 @@
+import ClockIcon from '~/icons/ClockIcon';
+
+import { Button } from '../core/Button/Button';
+import colors from '../core/colors';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseM, BaseS } from '../core/Text/Text';
+
+type Props = {
+  setIsEditMode: (editMode: boolean) => void;
+  queryRef: EmailVerificationStatusFragment$key;
+};
+
+function EmailVerificationStatus({ setIsEditMode, queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment EmailVerificationStatusFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              id
+            }
+            email {
+              email
+              verificationStatus
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const savedEmail = query?.viewer?.email?.email;
+  const verificationStatus = query?.viewer?.email?.verificationStatus;
+
+  const handleEditClick = useCallback(() => {
+    setIsEditMode(true);
+  }, [setIsEditMode]);
+
+  const [resendVerificationEmail] = usePromisifiedMutation<EmailVerificationStatusMutation>(graphql`
+    mutation EmailVerificationStatusMutation @raw_response_type {
+      resendVerificationEmail {
+        ... on ResendVerificationEmailPayload {
+          __typename
+        }
+        ... on ErrInvalidInput {
+          __typename
+        }
+      }
+    }
+  `);
+
+  const { pushToast } = useToastActions();
+
+  const handleResendClick = useCallback(async () => {
+    function pushErrorToast() {
+      pushToast({
+        autoClose: true,
+        message: `Something went wrong while sending a verification email. We are looking into it.`,
+      });
+    }
+    try {
+      const response = await resendVerificationEmail({ variables: {} });
+      if (response.resendVerificationEmail?.__typename !== 'ResendVerificationEmailPayload') {
+        pushErrorToast();
+        reportError(
+          `Could not save email address, typename was ${response.resendVerificationEmail?.__typename}`
+        );
+      } else {
+        pushToast({
+          autoClose: true,
+          message: `We've resent you an email to verify your email address.`,
+        });
+      }
+    } catch (error) {
+      pushErrorToast();
+    }
+  }, [pushToast, resendVerificationEmail]);
+
+  const resendEmailButton = useMemo(() => {
+    return (
+      <ResendEmail onClick={handleResendClick}>
+        <BaseS color={colors.shadow}>Resend verification email</BaseS>
+      </ResendEmail>
+    );
+  }, [handleResendClick]);
+
+  const verificationStatusIndicator = useMemo(() => {
+    switch (verificationStatus) {
+      case 'Unverified':
+        return (
+          <>
+            <ClockIcon />
+            <BaseS>Pending verification</BaseS>
+            {resendEmailButton}
+          </>
+        );
+      case 'Verified':
+        return (
+          <>
+            <CircleCheckIcon />
+            <BaseS>Verified</BaseS>
+          </>
+        );
+      case 'Failed':
```

Comment:
What does it mean to be in a `Failed` state? Like how do I fail an email verification? 

Also, what does the `Admin` state mean?

Reply:
`Failed` captures edge cases like if the email couldn't be delivered to the address. Something went wrong with the verification flow. 
We can improve how we handle it in the UI  by adding more nuanced remedy steps for users who end up in this state, but to start this at leasts let us flag users who encountered an issue somewhere in the flow.

this is what Benny said for what`Admin` does:  "Basically this would be a version of Verified that is used in place of Verified on dev and local so we can shoot emails to our admin emails."